### PR TITLE
added validation for checkoutForm

### PR DESCRIPTION
**Pull request by Galkevich-Natalia**

### CODE REVIEW

File: src/components/validation/validation.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    const messageName = document.getElementById('messageName');
+    const messagePhone = document.getElementById('messagePhone');
+    const messageEmail = document.getElementById('messageEmail');
+    const messageCity = document.getElementById('messageCity');
+    const messageStreet = document.getElementById('messageStreet');
+    const messageHouse = document.getElementById('messageHouse');
+    const messageApartment = document.getElementById('messageApartment');
+
+    if (!validateName(name)) {
+        messageName.textContent = 'Проверьте данные Name';
```
Comment (AnnMurr) [inline src/components/validation/validation.js]: your website in English
your error messagies in Russian

File: src/components/validation/validation.js [hunk 2]
```diff
@@ -110,25 +101,30 @@ function validateApartment() {
...
-    const body = document.querySelector('body');
+function validateForm() {
     const arrayCalls = [validateName(), validatePhone(), validateEmail(), validateCity(), validateStreet(), validateHouse(), validateApartment()];
+    const isValidForm = arrayCalls.every((item) => item === true);
 
-    const isValidResult = arrayCalls.every((item) => item === true)
+    return isValidForm;
+}
+
+export function checkValidation() {
```
Comment (AnnMurr) [inline src/components/validation/validation.js]: rename

File: src/components/validation/validation.js [hunk 3]
```diff
@@ -110,25 +101,30 @@ function validateApartment() {
...
 
+
 const checkoutModalBtnClose = document.querySelector('.checkout__modal-btn');
 checkoutModalBtnClose.addEventListener('click', closeCheckoutModal);
 
-function closeCheckoutModal(event) {
+function closeCheckoutModal() {
     const checkoutModal = document.querySelector('.checkout__modal');
     const overlayCheckout = document.querySelector('.overlayCheckout');
     const body = document.querySelector('body');
```
Comment (AnnMurr) [inline src/components/validation/validation.js]: checkValidation and  closeCheckoutModal mustn't be in validation file

**Review by AnnMurr — CHANGES_REQUESTED** [review]

File: src/components/checkout/checkout.js [hunk 4]
```diff
@@ -15,8 +21,29 @@ function getCheckoutTotalPrice() {
...
+
+export function closeCheckoutModal() {
+    const checkoutModal = document.querySelector('.checkout__modal');
+    const overlayCheckout = document.querySelector('.overlayCheckout');
+    const body = document.querySelector('body');
+
+    checkoutModal.remove();
+    overlayCheckout.style.display = 'none';
+    body.classList.remove('checkout__modal-open');
+
```
Comment (AnnMurr) [inline src/components/checkout/checkout.js]: you can remove these all

File: src/components/checkout/checkout.js [hunk 5]
```diff
@@ -15,8 +21,29 @@ function getCheckoutTotalPrice() {
     totalPriceValue.textContent = getFormatCurrency(checkoutTotalPrice);
 }
 
-const checkoutBtnOrder = document.querySelector('.checkout__block-btn');
-checkoutBtnOrder.addEventListener('click', checkValidation);
+function getOrder() {
+    if (validateForm()) {
```
Comment (AnnMurr) [inline src/components/checkout/checkout.js]: define validateForm() into a variable.

File: src/components/validation/validation.js [hunk 6]
```diff
@@ -8,7 +8,7 @@ function validateName() {
     const messageName = document.getElementById('messageName');
 
     if (!REGNAME.test(name)) {
-        messageName.textContent = 'Проверьте данные Name';
+        messageName.textContent = 'Please enter a valid name';
         messageName.style.color = "red";
         return false;
```
Comment (AnnMurr) [inline src/components/validation/validation.js]: you have the same code many times, so you can create  a separate function

Comment (AnnMurr) [reply src/components/validation/validation.js]: the same in else block

File: src/components/validation/validation.js [hunk 7]
```diff
@@ -1,10 +1,13 @@
+import {
+    REGNAME, REGPHONE, REGEMAIL,
+    REGCITY, REGSTREET, REGHOUSE, REGAPARTMENT
+} from "../../constants/constants";
+
 function validateName() {
     const name = document.getElementById('name').value;
```
Comment (AnnMurr) [inline src/components/validation/validation.js]: it's not just name it's name value

File: src/components/validation/validation.js [hunk 8]
```diff
@@ -18,9 +21,7 @@ function validatePhone() {
     const phone = document.getElementById('phone').value;
```
Comment (AnnMurr) [inline src/components/validation/validation.js]: the same

**Review by AnnMurr — APPROVED** [review]

File: src/components/checkout/checkout.js [hunk 9]
```diff
@@ -1,6 +1,13 @@
 import { getOrdersDataToStorage } from "../../utils/localStorage";
```
Comment (AnnMurr) [inline src/components/checkout/checkout.js:1]: kmk.

**Review by AnnMurr — CHANGES_REQUESTED** [review]

File: src/components/checkout/checkout.js (hunk 9 above)
Comment (AnnMurr) [inline src/components/checkout/checkout.js:1]: kllkm

**Review by AnnMurr — CHANGES_REQUESTED** [review]

File: src/components/checkout/checkout.js (hunk 9 above)
Comment (Galkevich-Natalia) [reply src/components/checkout/checkout.js:1]: ?

Comment (AnnMurr) [reply src/components/checkout/checkout.js:1]: don't pay attention

File: src/components/validation/validation.js [hunk 10]
```diff
@@ -4,103 +4,106 @@ import {
...
 
+function messageInvalidInput(messageValue, inputValue) {
+    messageValue.textContent = 'Please enter a valid ' + inputValue;
+    messageValue.style.color = "red";
+}
+
+function messsageValidInput(messageCorrectValue) {
+    messageCorrectValue.textContent = '';
+}
+
```
Comment (AnnMurr) [inline src/components/validation/validation.js]: can you move "return false/true" in these functions?

**Review by AnnMurr — CHANGES_REQUESTED** [review]

**Review by AnnMurr — APPROVED** [review]

**Pull request merged**